policy/modules/system/systemd.te

```diff
@@ -1417,3 +1417,516 @@ optional_policy(`
+# AVC avc:  denied  { rmdir } for  pid=1000 comm="systemd-homewor" name=".mozilla" dev="dm-14" ino=171140546 scontext=system_u:system_r:systemd_homed_t:s0 tcontext=unconfined_u:object_r:mozilla_home_t:s0 tclass=dir permissive=0
+# AVC avc:  denied  { write } for  pid=1000 comm="systemd-homewor" name=".mozilla" dev="dm-14" ino=171140546 scontext=system_u:system_r:systemd_homed_t:s0 tcontext=unconfined_u:object_r:mozilla_home_t:s0 tclass=dir permissive=0
+# AVC avc:  denied  { remove_name } for  pid=1000 comm="systemd-homewor" name="plugins" dev="dm-14" ino=18877888 scontext=system_u:system_r:systemd_homed_t:s0 tcontext=unconfined_u:object_r:mozilla_home_t:s0 tclass=dir permissive=0
+optional_policy(`
+	gen_require(`
+		type mozilla_home_t;
+	')
+	mozilla_read_user_home_files(systemd_homed_t)
+	allow systemd_homed_t mozilla_home_t:dir { remove_name rmdir write };
+')
+
+#### Allow: systemd-homework to create/handle /home/<username>.home file
+filetrans_pattern(systemd_homed_t, home_root_t, systemd_homed_home_file_t, file)
+manage_files_pattern(systemd_homed_t, home_root_t, systemd_homed_home_file_t)
+
+
+
+#
+# /run/systemd/userdb
+# link to other domains
+#
+
+#### Allow various subsystems to access /run/systemd/userdb/io.systemd.Home by having right context
+systemd_userdbd_runtime_filetrans(systemd_homed_t)
+#filetrans_pattern(systemd_homed_t, systemd_userdbd_runtime_t, systemd_userdbd_runtime_t, file, "io.systemd.Home")
+
+#### Allow: systemd-homed to manage /run/systemd/userdb/io.systemd.Home
+# AVC avc:  denied  { create } for pid=1000 comm="systemd-homed" name="io.systemd.Home" scontext=system_u:system_r:systemd_homed_t:s0 tcontext=system_u:object_r:systemd_userdbd_runtime_t:s0 tclass=sock_file permissive=1"
+systemd_manage_userdbd_runtime_sock_files(systemd_homed_t)
+
+#### Allow: various domains to connect /run/systemd/userdb/io.systemd.Home
+# IDK what this is because:
+# srw-rw-rw-. 1 root root system_u:object_r:systemd_userdbd_runtime_t:s0 0 10.10. 12:18 /run/systemd/userdb/io.systemd.Home
+# AVC avc:  denied  { connectto } for  pid=1000 comm="(bash)" path="/run/systemd/userdb/io.systemd.Home" scontext=system_u:system_r:init_t:s0 tcontext=system_u:system_r:systemd_homed_t:s0 tclass=unix_stream_socket permissive=0
+# AVC avc:  denied  { connectto } for  pid=1000 comm="systemd-user-ru" path="/run/systemd/userdb/io.systemd.Home" scontext=system_u:system_r:systemd_logind_t:s0 tcontext=system_u:system_r:systemd_homed_t:s0 tclass=unix_stream_socket permissive=0
+# AVC avc:  denied  { connectto } for  pid=1000 comm="systemd-userwor" path="/run/systemd/userdb/io.systemd.Home" scontext=system_u:system_r:systemd_userdbd_t:s0 tcontext=system_u:system_r:systemd_homed_t:s0 tclass=unix_stream_socket permissive=0
+systemd_homed_userdb_stream_connect(init_t)
+systemd_homed_userdb_stream_connect(systemd_logind_t)
+systemd_homed_userdb_stream_connect(systemd_userdbd_t)
+
```

Comment:
Yes so the above and the below should probably be addressed on a different level (nss?) because obviously maintaining these rules per domain individually is inefficient and hard to maintain.